### PR TITLE
Sc 1452 persist run combined cloud config json

### PR DESCRIPTION
**Pull request by blackboxsw**
## Proposed Commit Message
```
instance-data: write /run/cloud-init/merged-cloud-config.json

Persist the merged_cloud_config key as a new file to aid in
discovery or provided configuration settings.

The file /run/cloud-init/merged-cloud-config.json represents the
full configuration presented to cloud-init which layers config
settings in the following priority order:
1. default config: cloudinit.settings.CFG_BUILTIN
2. merged_system_cfg: config from file-system /etc/cloud and
    /etc/cloud.cfg.d
3. cloud-specific metadata
4. cloud-specific vendor-data, vendor-data2
5. cloud-specific user-data
    
Configuration keys present in earlier sources are overridden by
keys in later sources.
```

## Checklist:

 - [x] My code follows the process laid out in [the documentation](https://cloudinit.readthedocs.io/en/latest/development/contributing.html)
 - [x] I have updated or added any unit tests accordingly
 - [ ] I have updated or added any documentation accordingly

### CODE REVIEW

**Comment by dbungert** [comment]
Testing notes:

* using this branch, staged into the Subiquity snap, already fixes https://bugs.launchpad.net/subiquity/+bug/2022102 - though I don't know if that's the result of this branch directly or the branch it's based upon

* using this branch, built with `./packages/bddeb -S -d --release jammy` then using any suitable build strategy for a binary package, and staging into the host image of the mantic-live-server ISO, indeed produces the desired JSON at `/run/cloud-init/combined-cloud-config.json`.  Thanks!

**Comment by dbungert** [comment]
I created a proof of concept branch to consume the data from this location.  https://github.com/dbungert/subiquity/pull/new/consume-combined-cloud-config

The change is straightforward and works nicely .

File: tests/unittests/test_data.py
Comment (aciba90) [inline tests/unittests/test_data.py:1]: Thanks for migrating this to pytest.

**Review by aciba90 — CHANGES_REQUESTED** [review]
Overall looks great thanks. Nice touch moving functions to `atomic_helper.py`.

There are a couple of warnings in `tests/integration_tests/modules/test_jinja_templating.py::test_substitution_in_etc_cloud` which need some caring. I think they are tests problems, but I didn't dive into them.

**Comment by blackboxsw** [comment]
> There are a couple of warnings in `tests/integration_tests/modules/test_jinja_templating.py::test_substitution_in_etc_cloud` which need some caring. I think they are tests problems, but I didn't dive into them.

Yes, the value we were sourcing in this integration test in `echo {{merged_cfg._doc}}` now has a colon in the prefix `DEPRECATED: `  which lead to interpreting the jinja rendered content as invalid #cloud-config YAML dict instead of list.

I've used the intended not-deprecated key `merged_system_cfg` which doesn't cause this testing problem.

```
87.42s setup    tests/integration_tests/modules/test_jinja_templating.py::test_substitution_in_etc_cloud
11.60s call     tests/integration_tests/modules/test_jinja_templating.py::test_substitution_in_etc_cloud
0.65s teardown tests/integration_tests/modules/test_jinja_templating.py::test_substitution_in_etc_cloud
=================== 1 passed, 2 warnings in 99.67s (0:01:39) ===================
___________________________________ summary ____________________________________
  integration-tests: commands succeeded
  congratulations :)
```

File: tests/integration_tests/modules/test_jinja_templating.py [hunk 1]
```diff
@@ -52,7 +52,7 @@ def test_substitution_in_etc_cloud(client: IntegrationInstance):
     new_cloud_part = (
         "## template: jinja\n"
         "bootcmd:\n"
-        " - echo {{merged_cfg._doc}} > /var/tmp/bootcmd_output\n"
+        " - echo '{{merged_system_cfg._doc}}' > /var/tmp/bootcmd_output\n"
```
Comment (TheRealFalcon) [inline tests/integration_tests/modules/test_jinja_templating.py:55]: Same change needed on line 20 too

**Review by TheRealFalcon — APPROVED** [review]
One more integration test update needed, but I'll trust you can apply that on your own. Other than that, +1 from me.

**Review by aciba90 — APPROVED** [review]

**Pull request merged**